ThisSensorManager-back.php

```diff
@@ -0,0 +1,469 @@
+                case 1:
+                    if($state[key($state)] == self::REWORK)
+                    {
+                        $stop = $this->database->table($sName)->where("id=?", array_key_first($ids[$id]))->fetch()->time->getTimestamp();
+                        $sState = 0;
+                        $time += $stop-$start;
+                        //echo " -> REWORK -> ".$stop."-> ALL: ". $time;
+                    }
+                    break;
+
+                // default:
+                //     # code...
+                //     break;
+            }
+            //echo "<br><br>";
+        }        
+
+        // $first = $this->database->table($sName)->where("id=?", $ids[0])->fetch()->time->getTimestamp();
+        // $last = $this->database->table($sName)->where("id=?", end($ids))->fetch()->time->getTimestamp();
+        // $res =  $last-$first;
+        return array(new DateInterval("PT".$time."S"), $time);
+
+    }
+
+    public function getAproxStopTime($stopTimestamp, $stopCount)
+    {
+        $time = ceil($stopTimestamp/$stopCount);
+        return array(new DateInterval("PT".$time."S"), $time);
+    }    
+
+    public function getWorkTime($allTimestamp, $stopTimestamp)
+    {
+        $time = $allTimestamp-$stopTimestamp;
+        return array(new DateInterval("PT".$time."S"), $time);
+    }
+
+
+
+    public function getAproxWorkTime($workTimestamp, $workCount)
+    {
+        $time = ceil($workTimestamp/$workCount);
+        return array(new DateInterval("PT".$time."S"), $time);
+    }
+
+    public function getWorkTimeOld($sName, $ids): DateInterval
+    {
+        $last = $this->database->table($sName)->where("id=?", 1)->fetch();
+        //echo $out =  $last->time->getTimestamp();
+        //dump($last->time);
+        $workTime = 0;
+
+        // $next = $this->database->table($sName)->where("id=?", 2)->fetch();
+        // //echo $next->time->getTimestamp();
+        // dump($next->time);
+
+
+        // $start = DateTime::from($out);
+
+        // //echo $start;
+
+
+        foreach($ids as $id)
+        {
+            $actual = $this->database->table($sName)->where("id=?", $id)->fetch();
+
+            //dump($last->actWork);
+
+            //  First loop
+            if($id < $ids[1])
+            {
+                $actWork=3*self::MINUTE; // 3 Minutes
+                // $workTime+=$actWork;
+                $lastWork = $actWork;
+                // $this->database->table($sName)->where("id=?", $id)->update(['work'=>$actWork, 'time'=>$actual->time]);
+                //echo "FIRST";
+
+                //echo " -ID:".$id." Last: ".$last->time." Actual: ".$actual->time."<br>";
+                //echo "Add:".$actWork."->T:".gmdate("H:i:s",$actWork)." -> result: ".$workTime."->T:".gmdate("H:i:s",$workTime)."<br><br><br>";
+
+
+            }
+            // Next loops
+            else if($id >= $ids[1])
+            {
+                $actWork=$actual->time->getTimestamp()-$last->time->getTimestamp();
```

Comment:
1. Do you use some code format? This is not well formated. You can set autoformat after each Save file in VS Code and probably also in PHPStorm.
2. Without the correct formatting, it wasn't clear for me what this code does (the `-` between `getTimestamp()` and `$last` is not soo obvious and I thought that it is `->`.